src/overlays/actors/ovl_En_Ishi/z_en_ishi.c

```diff
@@ -21,47 +55,420 @@ const ActorInit En_Ishi_InitVars = {
+    static u16 liftSounds[] = { NA_SE_PL_PULL_UP_ROCK, NA_SE_PL_PULL_UP_BIGROCK };
+    s32 pad;
+    s16 type = this->actor.params & 1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ishi/func_80A7ED60.s")
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        EnIshi_SetupLiftedUp(this);
+        Audio_PlaySoundAtPosition(globalCtx, &this->actor.posRot.pos, 20, liftSounds[type]);
+        if ((this->actor.params >> 4) & 1) {
+            EnIshi_SpawnBugs(this, globalCtx);
+        }
+    } else if (this->collider.base.acFlags & 2 && (type == ROCK_SMALL) &&
+               this->collider.body.acHitItem->toucher.flags & 0x40000048) {
+        EnIshi_DropCollectible(this, globalCtx);
+        Audio_PlaySoundAtPosition(globalCtx, &this->actor.posRot.pos, sBreakSoundDurations[type], sBreakSounds[type]);
+        sFragmentSpawnFuncs[type](this, globalCtx);
+        sDustSpawnFuncs[type](this, globalCtx);
+        Actor_Kill(&this->actor);
+    } else if (this->actor.xzDistFromLink < 600.0f) {
+        Collider_CylinderUpdate(&this->actor, &this->collider);
+        this->collider.base.acFlags &= ~2;
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        if (this->actor.xzDistFromLink < 400.0f) {
+            CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+            if (this->actor.xzDistFromLink < 90.0f) {
+                if (type == ROCK_LARGE) {
+                    func_8002F434(&this->actor, globalCtx, 0, 80.0f, 20.0f);
+                } else {
+                    func_8002F434(&this->actor, globalCtx, 0, 50.0f, 10.0f);
+                }
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ishi/func_80A7ED94.s")
+void EnIshi_SetupLiftedUp(EnIshi* this) {
+    this->actionFunc = EnIshi_LiftedUp;
+    this->actor.room = -1;
+    this->actor.flags |= 0x10;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ishi/func_80A7EE1C.s")
+void EnIshi_LiftedUp(EnIshi* this, GlobalContext* globalCtx) {
+    if (Actor_HasNoParent(&this->actor, globalCtx)) {
+        this->actor.room = globalCtx->roomCtx.curRoom.num;
+        if ((this->actor.params & 1) == ROCK_LARGE) {
+            Flags_SetSwitch(globalCtx, ((this->actor.params >> 0xA) & 0x3C) | ((this->actor.params >> 6) & 3));
+        }
+        EnIshi_SetupFly(this);
+        EnIshi_Fall(this);
+        func_80A7ED94(&this->actor.velocity, D_80A7FA28[this->actor.params & 1]);
+        func_8002D7EC(&this->actor);
+        func_8002E4B4(globalCtx, &this->actor, 7.5f, 35.0f, 0.0f, 0xC5);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ishi/EnIshi_Init.s")
+void EnIshi_SetupFly(EnIshi* this) {
+    this->actor.velocity.x = Math_Sins(this->actor.posRot.rot.y) * this->actor.speedXZ;
+    this->actor.velocity.z = Math_Coss(this->actor.posRot.rot.y) * this->actor.speedXZ;
+    if (!(this->actor.params & 1)) {
```

Comment:
this is checking the type so it should be `== ROCK_SMALL`

Reply:
Well TIL the reason I couldn't get it to work is because equals has a higher priority than bitwise and lmao. It's good now.